### ISSUE TITLE
Feature Request: Use the 'getInstance(defaultInstance)' method from the base class (Typescript client)

### ISSUE THREAD

**Issue by Mrks83** [RicoSuter/NSwag#3688]
Currently there is an option to "getBaseUrl" from base class, which is fine. It would be awesome to get the same for the second constructor parameter `instance`. That way interceptors can easily be added and constructor-calls with `null` on the first parameter `new MyClient(null, myInstance)` are not needed.

```
    constructor(baseUrl?: string, instance?: AxiosInstance) {
        super();
        this.instance = instance ? instance : axios.create();
        this.baseUrl = baseUrl !== undefined && baseUrl !== null ? baseUrl : this.getBaseUrl("https://localhost");
    }
```

**Comment by BoberPL** [comment]
Yeah, that would be really awesome. Sadly this issue has been probably lost in tons of issues that are raised here.
It would really help not having to pass the instance to which interceptors are wired up each time we are using a client, but use similar approach like with baseUrl.
`new MyClient()`.... how beautiful that would be.
